### PR TITLE
Prepare release 2.10.1

### PR DESCRIPTION
**Pull request by joelsmith**
This PR adds support for features added in the operands in 2.10.x and makes all necessary changes to be ready for the 2.10.1 release, specifically:

* Move ServiceMonitor and PodMonitor objects to separate resource file
   Because the release process overwrites the `keda.yaml` file, we need resources which are introduced in this repo to live in a separate file so that they won't be overwritten and lost each time the the release happens.
* Use version sort when deciding which manifests to diff
   As this is the first 2.10.x release, we noticed that the manifest diff was looking at the wrong change set.
* Add GRPC certs volume for metrics server to match operand
   The metrics server now makes a GRPC connection to the main KEDA controller, so it needs access to the certificates used for that connection
* Add admission webhooks support 
  2.10.X adds a new controller and a deployment to run it. The operator now supports deploying it
* Prepare release 2.10.1
  All release process items to do the 2.10.1 release

### Checklist

- [X] Commits are signed with Developer Certificate of Origin (DCO)

### CODE REVIEW

**Comment by zroubalik** [comment]
> It appears that there are some changes in the operand repo's manifests which need to be taken into account here:
> 
>     * The `keda-operator` seems to want to create a secret with certs in it and requires the following (or equivalent privs):
> 
> 
> ```
> ---
> apiVersion: rbac.authorization.k8s.io/v1
> kind: Role
> metadata:
>   name: foo
>   namespace: keda
> rules:
> - apiGroups:
>   - ""
>   resources:
>   - secrets
>   verbs:
>   - create
>   - get
>   - update
> ---
> apiVersion: rbac.authorization.k8s.io/v1
> kind: RoleBinding
> metadata:
>   name: foo
>   namespace: keda
> roleRef:
>   apiGroup: rbac.authorization.k8s.io
>   kind: Role
>   name: foo
> subjects:
> - kind: ServiceAccount
>   name: keda-operator
>   namespace: keda
> ```
> 
>     * It appears that the olm operator is trying to inject a `/certs` volume into the metrics server deployment, but it seems that the new version of the deployment already has that volume and the two are conflicting.

you are right, this is a new change introduced in 2.10 upstream

File: go.mod [hunk 1]
```diff
@@ -1,89 +1,80 @@
...
-	github.com/onsi/gomega v1.24.1
+	github.com/onsi/gomega v1.27.6
 	github.com/openshift/api v3.9.0+incompatible
-	k8s.io/api v0.25.4
-	k8s.io/apimachinery v0.25.4
-	k8s.io/apiserver v0.25.0
-	k8s.io/client-go v0.25.4
-	k8s.io/kube-aggregator v0.24.3
-	sigs.k8s.io/controller-runtime v0.13.1
+	k8s.io/api v0.26.3
```
Comment (zroubalik) [inline go.mod:12]: I think we can bump k8s deps to 0.26.4 - this version wasn't present when we made 2.10.1 upstream.

**Comment by gauron99** [comment]
thx for the ping. I see what it is. its because "2.10" is alphabetically not following "2.9". I made the script to compare the 2 last dirs but I think Im gonna have to parse the number and then take 2 last ones. Ill make a fix

**Comment by jrvaldes** [comment]
> thx for the ping. I see what it is. its because "2.10" is alphabetically not following "2.9". I made the script to compare the 2 last dirs but I think Im gonna have to parse the number and then take 2 last ones. Ill make a fix

@gauron99  could we leverage any existing library/tool to compare sem versions?

File: resources/keda-olm-operator.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+---
```
Comment (jrvaldes) [inline resources/keda-olm-operator.yaml:1]: Consider moving this commit to a separate PR, as the changes are unrelated to the release prep.

File: go.mod (hunk 1 above)
Comment (joelsmith) [reply go.mod]: Thanks, done

**Review by zroubalik — COMMENTED** [review]
Looking good! The only minor thing is a Readme, could you please update the example KedaController object with the new admission webhook config?
https://github.com/kedacore/keda-olm-operator/blob/main/README.md#kedacontroller-spec
